src/reactionsystem.jl

```diff
@@ -0,0 +1,176 @@
+using Catalyst, ModelingToolkit, Symbolics
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(model::Model; kwargs...)  # Todo: requires unique parameters (i.e. SBML must have been imported with localParameter promotion in libSBML)
+    checksupport(model)
+    model = make_extensive(model)
+    rxs = mtk_reactions(model)
+    t = Catalyst.DEFAULT_IV
+    species = [create_var(k) for k in keys(model.species)]
+    params = vcat([create_param(k) for k in keys(model.parameters)], [create_param(k) for k in keys(model.compartments)])
+    ReactionSystem(rxs,t,species,params; kwargs...)
+end
+
+""" ReactionSystem constructor """
+function ModelingToolkit.ReactionSystem(sbmlfile::String; kwargs...)
+    model = readSBML(sbmlfile)
+    ReactionSystem(model; kwargs...)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(model::Model; kwargs...)
+    rs = ReactionSystem(model; kwargs...)
+    model = make_extensive(model)  # PL: consider making `make_extensive!` to avoid duplicate calling in ReactionSystem and here
+    u0map = get_u0(model)
+    parammap = get_paramap(model)
+    defaults = Dict(vcat(u0map, parammap))
+    convert(ODESystem, rs, defaults=defaults)
+end
+
+""" ODESystem constructor """
+function ModelingToolkit.ODESystem(sbmlfile::String; kwargs...)
+    model = readSBML(sbmlfile)
+    ODESystem(model; kwargs...)
+end
+
+""" ODEProblem constructor """
+function ModelingToolkit.ODEProblem(model::Model,tspan;kwargs...)  # PL: Todo: add u0 and parameters argument
+    odesys = ODESystem(model)
+    ODEProblem(odesys, [], tspan; kwargs...)
+end
+
+""" ODEProblem constructor """
+function ModelingToolkit.ODEProblem(sbmlfile::String,tspan;kwargs...)  # PL: Todo: add u0 and parameters argument
+    odesys = ODESystem(sbmlfile)
+    ODEProblem(odesys, [], tspan; kwargs...)
+end
+
+""" Check if conversion to ReactionSystem is possible """
+function checksupport(model)
+    # for reaction in model.reactions
+    #     if reaction.reversible
+    #         throw(DomainError(reaction, "Cannot convert `kineticLaw` of reversible reaction to MTK.Reaction.rate"))
+    #     end
+    # end
+    return  # Todo: convert all Reactions that are `reversible=true` to a forward and reverse reaction with `reversible=false`.
+end
+
+""" Convert intensive to extensive expressions """
+function make_extensive(model)
+    model = to_initial_amounts(model)
+    model = to_extensive_math!(model)
+    model
+end
+
+""" Convert initial_concentration to initial_amount """
+function to_initial_amounts(model::Model)
+    model = deepcopy(model)
```

Comment:
you don't need a full copy here, just copy()ing the model struct and deepcopying the species is okay.